src/Core/Settings.cpp

```diff
@@ -5455,6 +5455,14 @@ Only available in ClickHouse Cloud. Number of background threads for speculative
 )", 0) \
     DECLARE(Int64, ignore_cold_parts_seconds, 0, R"(
 Only available in ClickHouse Cloud. Exclude new data parts from SELECT queries until they're either pre-warmed (see cache_populated_by_fetch) or this many seconds old. Only for Replicated-/SharedMergeTree.
+)", 0) \
+    DECLARE(Bool, short_circuit_default_implementation_for_nulls, true, R"(
+Setting for short-circuit default implementations for nulls in function with useDefaultImplementationForNulls() = true.
+If true, function will not actually evaluate for rows in which there are at least one argument with null value.
+)", 0) \
+    DECLARE(Double, short_circuit_default_implementation_for_nulls_threshold, 1.0, R"(
+Ratio threshold for short-circuit default implementations for nulls in function with useDefaultImplementationForNulls() = true. It is only valid when short_circuit_default_implementation_for_nulls is true.
+When the ratio of rows containing nulls to the total number of rows exceeds this threshold, these rows containing nulls would not be evaluated.
```

Comment:
From user perspective it's unclear what is `default implementation for nulls`, let's not use it in setting name and description and write better description.
For setting names, maybe better to use smth like `short_circuit_function_evaluation_for_nulls` and `short_circuit_function_evaluation_for_nulls_threshold`